debezium-core/src/main/java/io/debezium/pipeline/source/spi/SnapshotChangeEventSource.java

```diff
@@ -27,4 +27,11 @@
      *             in case the snapshot was aborted before completion
      */
     SnapshotResult execute(ChangeEventSourceContext context) throws InterruptedException;
+
+    /**
+     * Set a listener that is invoked for important events that happens during snapshot process, like start, stop, table completed etc.
+     *
+     * @param listener
+     */
+    public void setSnapshotProgressListener(SnapshotProgressListener listener);
```

Comment:
Hum hum, I wished we could avoid this so the listener reference in the implementation can be final.